src/scene/camera.cpp

```diff
@@ -56,25 +54,30 @@ void SceneCamera::setup_listeners() {
     });
 }
 
-void SceneCamera::recalculate() {
+void Camera::recalculate() {
     V = glm::lookAt(cam_pos, cam_pos + cam_front, cam_up);
 }
 
-void SceneCamera::reset() {
+void Camera::reset() {
     cam_pos = default_pos;
     cam_front = default_front;
     cam_up = default_up;
     recalculate();
 }
 
-void SceneCamera::displace_cam(glm::vec3 displacement) {
+void Camera::displace_cam(glm::vec3 displacement) {
     cam_pos = cam_pos + displacement;
     recalculate();
 }
 
+// Called during update loop
+void Camera::update() {
```

Comment:
We can do smoother camera movement here instead of movement only on joystick input. @AustinPuk please fill this function out.